ragelo/llm_providers/ollama_client.py

```diff
@@ -33,19 +34,35 @@ def __call__(
 
         Args:
             prompt: The prompt to use. Either a list of messages or a string.
-            temperature : The temperature to use. Defaults to 0.1.
-            max_tokens: The maximum number of tokens to retrieve. Defaults 512.
         """
+
+        def run(coroutine):
+            return asyncio.run(coroutine)
+
         if isinstance(prompt, str):
             prompt = [{"role": "system", "content": prompt}]
-        answers = asyncio.run(
-            self.__ollama_client.chat.completions.create(
-                model=self.config.model,
-                messages=prompt,  # type: ignore
-                temperature=self.config.temperature,
-                max_tokens=self.config.max_tokens,
+        try:
```

Comment:
what was the problem of the previous approach?

Reply:
It would not work if you were calling the LLMProvider directly from a notebook. This is similar to what we do in the other evaluators when calling `batch_evaluate`